modules/tile-converter/src/i3s-converter/attribute-metadata-info.ts

```diff
@@ -0,0 +1,246 @@
+      header: [{property: 'count', valueType: 'UInt32'}],
+      attributeValues: {valueType: 'Int32', valuesPerElement: 1}
+    };
+
+    switch (attributeType) {
+      case AttributeType.OBJECT_ID_TYPE:
+        this.setupIdAttribute(storageAttribute);
+        break;
+      case AttributeType.STRING_TYPE:
+        this.setupStringAttribute(storageAttribute);
+        break;
+      case AttributeType.DOUBLE_TYPE:
+        this.setupDoubleAttribute(storageAttribute);
+        break;
+      case AttributeType.SHORT_INT_TYPE:
+        break;
+      default:
+        this.setupStringAttribute(storageAttribute);
+    }
+
+    return storageAttribute;
+  }
+
+  /**
+   * Finds and returns attribute type based on key form propertyTable.
+   * @param attributeType
+   */
+  private getFieldAttributeType(attributeType: Attribute): ESRIField {
+    switch (attributeType) {
+      case AttributeType.OBJECT_ID_TYPE:
+        return 'esriFieldTypeOID';
+      case AttributeType.STRING_TYPE:
+        return 'esriFieldTypeString';
+      case AttributeType.SHORT_INT_TYPE:
+        return 'esriFieldTypeInteger';
+      case AttributeType.DOUBLE_TYPE:
+        return 'esriFieldTypeDouble';
+      default:
+        return 'esriFieldTypeString';
+    }
+  }
+
+  /**
+   * Sets up Id attribute for map segmentation.
+   * @param storageAttribute - attribute for map segmentation .
+   */
+  private setupIdAttribute(storageAttribute: AttributeStorageInfo): void {
+    storageAttribute.attributeValues = {
+      valueType: 'Oid32',
+      valuesPerElement: 1
+    };
+  }
+
+  /**
+   * Sets up storage attribute as string.
+   * @param storageAttribute - attribute for map segmentation.
+   */
+  private setupStringAttribute(storageAttribute: AttributeStorageInfo): void {
+    // @ts-expect-error
+    storageAttribute.ordering.unshift('attributeByteCounts');
+    storageAttribute.header.push({property: 'attributeValuesByteCount', valueType: 'UInt32'});
+    storageAttribute.attributeValues = {
+      valueType: 'String',
+      encoding: 'UTF-8',
+      valuesPerElement: 1
+    };
+    storageAttribute.attributeByteCounts = {
+      valueType: 'UInt32',
+      valuesPerElement: 1
+    };
+  }
+
+  /**
+   * Sets up double attribute for map segmentation.
+   * @param storageAttribute - attribute for map segmentation .
+   */
+  private setupDoubleAttribute(storageAttribute: AttributeStorageInfo): void {
+    storageAttribute.attributeValues = {
+      valueType: 'Float64',
+      valuesPerElement: 1
+    };
+  }
+
+  /**
+   * Sets up field attribute for map segmentation.
+   * @param key - attribute for map segmentation.
+   * @param fieldAttributeType - esri attribute type ('esriFieldTypeString' or 'esriFieldTypeOID').
+   */
+  private createFieldAttribute(key: string, fieldAttributeType: ESRIField): Field {
+    return {
+      name: key,
+      type: fieldAttributeType,
+      alias: key
+    };
+  }
+
+  /**
+   * Generates popup info to show metadata on the map.
```

Comment:
I believe in strict separation between loading / logic and presentation layers. talking about "popup" (which is very much a presentation layer concept) here in the load doesn't sit right with me. 

Either rename this to something non-UI-related, or maybe move such functionality to the application